### PR TITLE
DM-38703: Add missing bright star feature to subtractBrightStars

### PR DESCRIPTION
**Pull request by leeskelvin**

### CODE REVIEW

File: python/lsst/meas/algorithms/brightStarStamps.py [hunk 1]
```diff
@@ -367,13 +372,57 @@ def initAndNormalize(
...
+                            )
+
+                        except RuntimeError:
+                            stamp.annularFlux = np.nan
+                            logger.error(
+                                "The annulux flux was not found for radii {} and {}".format(
+                                    newInnerRadius, newOuterRadius
+                                )
+                            )
+                        if stamp.annularFlux and stamp.annularFlux > 0:
```
[the remark this answers is not in the document]
Comment (bazkiaei) [reply python/lsst/meas/algorithms/brightStarStamps.py:404]: The if block at line [381](https://github.com/lsst/meas_algorithms/pull/344/files#diff-1561000694a576a34446570d3a89d32b0528de32ac026fc5d0decce768b1dd38R381) does that.

File: python/lsst/meas/algorithms/brightStarStamps.py [hunk 2]
```diff
@@ -367,13 +372,57 @@ def initAndNormalize(
...
+                                newAnnulus,
+                                statsControl=statsControl,
+                                statsFlag=statsFlag,
+                                badMaskPlanes=badMaskPlanes,
+                            )
+
+                        except RuntimeError:
+                            stamp.annularFlux = np.nan
+                            logger.error(
+                                "The annulux flux was not found for radii {} and {}".format(
```
[the remark this answers is not in the document]
Comment (bazkiaei) [reply python/lsst/meas/algorithms/brightStarStamps.py:400]: corrected

File: python/lsst/meas/algorithms/brightStarStamps.py [hunk 3]
```diff
@@ -367,13 +372,57 @@ def initAndNormalize(
...
+                                    newInnerRadius, newOuterRadius
+                                )
+                            )
+                        if stamp.annularFlux and stamp.annularFlux > 0:
+                            logger.info("The flux is found within an optimized annulus.")
+                            logger.info(
+                                "The optimized annulus raddi are {} and {} and the flux is {}".format(
```
[the remark this answers is not in the document]
Comment (bazkiaei) [reply python/lsst/meas/algorithms/brightStarStamps.py:407]: corrected

[the remark this answers is not in the document]
Comment (timj) [reply python/lsst/meas/algorithms/brightStarStamps.py:407]: Policy is not use format in log messages but to use deferred formatting using %s etc.

[the remark this answers is not in the document]
Comment (leeskelvin) [reply python/lsst/meas/algorithms/brightStarStamps.py]: All loggers switched to using deferred formatting.

File: python/lsst/meas/algorithms/brightStarStamps.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (leeskelvin) [reply python/lsst/meas/algorithms/brightStarStamps.py:421]: Added an `if` check in the `initAndNormalize` method which fails fast if the annulus width is less than 1 pixel:
```
if annulusWidth < 1:
    raise ValueError("The annulus width must be greater than 1 pixel.")
```

**Review by taranu — APPROVED** [review]

**Pull request merged**